### PR TITLE
Fixes #25249 - update Rubocop to 0.50.x

### PR DESCRIPTION
**Pull request by mmoll**
this is the last version supporting Ruby 2.0

### CODE REVIEW

**Comment by mmoll** [comment]
The new .rubocop.yml merges the previous disablings with the ones we have in core.

File: bundler.d/test.rb [hunk 1]
```diff
@@ -6,7 +6,6 @@
   gem 'public_suffix', '< 3' if RUBY_VERSION < '2.1'
   gem 'benchmark-ips'
   gem 'ruby-prof'
-  gem 'rubocop', '0.38.0'
```
Comment (ekohl) [inline bundler.d/test.rb:9]: Won't this be an issue for jenkins?

Comment (mmoll) [reply bundler.d/test.rb:9]: I didn't think so, but now that I checked it, yes... In core it's implemented differently, but anyway I'll revert this.

Comment (lzap) [reply bundler.d/test.rb:9]: What kind of issue? I don't understand.

Comment (ekohl) [reply bundler.d/test.rb:9]: Now that it's back in test.rb you probably can't see the full context but it was moved to development.rb. IIRC Jenkins does `bundle install --without development` and not have rubocop installed.

**Review by lzap — APPROVED** [review]
Let's merge and see thanks.

**Pull request merged**